### PR TITLE
Fix issue with pathless config items

### PR DESCRIPTION
**Pull request by cecinestpasunepipe**

### CODE REVIEW

File: dissect/target/plugins/os/unix/etc/etc.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def _sub(self, items: ConfigurationEntry, entry: Path, pattern: str) -> Iterator
             if fnmatch.fnmatch(path, pattern):
                 data = {
                     "_target": self.target,
-                    "source": self.target.fs.path(config_entry.entry.path),
+                    "source": self.target.fs.path(config_entry.entry.path if hasattr(config_entry,"entry") else path),
```
[the remark this answers is not in the document]
Comment (cecinestpasunepipe) [reply dissect/target/plugins/os/unix/etc/etc.py:51]: Can you give an example regarding the first comment?

[the remark this answers is not in the document]
Comment (cecinestpasunepipe) [reply dissect/target/plugins/os/unix/etc/etc.py]: Ah thanks

File: dissect/target/plugins/os/unix/etc/etc.py [hunk 2]
```diff
@@ -28,9 +28,10 @@ class EtcTree(ConfigurationTreePlugin):
     def __init__(self, target: Target):
         super().__init__(target, "/etc")
 
-    def _sub(self, items: ConfigurationEntry, entry: Path, pattern: str) -> Iterator[UnixConfigTreeRecord]:
+    def _sub(
+        self, items: Union[ConfigurationEntry, dict], entry: Path, orig_path: Path, pattern: str
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/etc/etc.py:32]:
```suggestion
        self, items: ConfigurationEntry | dict, entry: Path, orig_path: Path, pattern: str
```

File: dissect/target/plugins/os/unix/etc/etc.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import fnmatch
 import re
 from pathlib import Path
-from typing import Iterator
+from typing import Iterator, Union
```
Comment (Schamper) [inline dissect/target/plugins/os/unix/etc/etc.py:4]:
```suggestion
from typing import Iterator
```

**Review by Schamper — CHANGES_REQUESTED** [review]

**Pull request merged**